main.js

```diff
@@ -346,48 +354,56 @@ function readValue(port) {
         if (err) {
             adapter.log.error('Cannot read port ' + port + ': ' + err);
         } else {
-            adapter.setState('gpio.' + port + '.state', !!value, true);
+            adapter.setState('gpio.' + port + '.state', inputPullUp(value), true);
         }
     });
 }
 
 function syncPort(port, data, callback) {
+    data.isButton = (data.input === 'button');
+    data.isInput = (data.input === 'in' || data.input === 'true' || data.input === true || data.isButton);
     adapter.getObject('gpio.' + port + '.state', (err, obj) => {
-        if (data.enabled) {
-            let isInput = (data.input === 'in' || data.input === 'true' || data.input === true);
-
+        if (data.enabled && !data.isButton) {
             if (err || !obj || !obj.common) {
                 obj = {
                     common: {
                         name:  'GPIO ' + port,
                         type:  'boolean',
-                        role:  isInput ? 'indicator' : 'switch',
-                        read:  isInput,
-                        write: !isInput
+                        role:  data.isInput ? 'indicator' : 'switch',
+                        read:  data.isInput,
+                        write: !data.isInput
                     },
                     native: {
                     },
                     type: 'state'
                 };
-                adapter.setObject('gpio.' + port + '.state', obj, () =>
-                    syncPortDirection(port, data, callback));
+                adapter.setObject('gpio.' + port + '.state', obj, () => {
```

Comment:
When we do a bigger update we should replace setObject by extendObject

Reply:
Well... I just followed the lead of what was already there for consistency, but OK, will update.